subprojects/core/src/main/java/org/gradle/api/initialization/Settings.java

```diff
@@ -208,4 +209,20 @@
      */
     @Incubating
     void buildCache(Action<? super BuildCacheConfiguration> action);
+
+    /**
+     * Configures plugin management.
+     *
+     * @since 3.5
+     */
+    @Incubating
+    void pluginManagement(Action<? super PluginManagementSpec> pluginManagementSpec);
```

Comment:
I wonder if we really need this method (and these `dslTarget(Action<? super DslTarget> spec)` methods in general) when there's already a getter and this can be implemented with zero runtime cost at the Kotlin side with an inline `invoke` method.

We could introduce some marker interface or annotation that would let the runtime know this type `PluginManagementSpec` should be decorated.